File: scripts/test_solutions/621_task_scheduler.py
def leastInterval(tasks, n):
    freq = [0] * 26
    for t in tasks:
        freq[ord(t) - 65] += 1
    max_freq = max(freq)
    max_count = sum(1 for f in freq if f == max_freq)
    return max(len(tasks), (max_freq - 1) * (n + 1) + max_count)

File: scripts/test_solutions/621_task_scheduler.js
var leastInterval = function(tasks, n) {
    const freq = Array(26).fill(0);
    for (const t of tasks) freq[t.charCodeAt(0) - 65]++;
    const maxFreq = Math.max(...freq);
    const maxCount = freq.filter(f => f === maxFreq).length;
    return Math.max(tasks.length, (maxFreq - 1) * (n + 1) + maxCount);
};
